engine/utils/collections/StackArray.h

```diff
@@ -0,0 +1,480 @@
+     */
+    inline constexpr const size_t Size() const
+    {
+        return S;
+    }
+
+    /**
+     * @brief Returns the size of the array in bytes.
+     * @return the size of the array in bytes.
+     */
+    inline const size_t Count() const
+    {
+        return count;
+    }
+
+    // Setters
+
+    /**
+     * @brief Inserts a given element into the position specified by `index`. If the provided index
+     * is larger than the array's size, an exception will be thrown.
+     * @param index the position in the array we want to insert the element into.
+     * @param element the element we want to store in the array.
+     */
+    inline void Insert(const size_t& index, const T& element)
+    {
+        count += !bitField.IsSet(index + 1);
+        bitField.SetBit(index + 1);
+        Set(index, element);
+    }
+
+    inline void Append(const T& element)
+    {
+        ArrayUtils::AssertIsInBounds(count, S);
+        Set(count, element);
+        count += !bitField.IsSet(count + 1);
+        bitField.SetBit(count);
+    }
+
+    /**
+     * Checks if the MSArray is empty.
+     * @return true if the MSArray has had no allocations made, false if at least one element was
+     * added.
+     */
+    inline bool Empty() const
+    {
+        return count == 0;
+    }
+
+    /**
+     * Checks if a specified index has been assigned to. If the provided index
+     * is larger than the array's size, an exception will be thrown.
+     * @param index the index of the element being searched for.
+     * @return true if the index has been previously assigned to.
+     */
+    inline bool Active(const size_t& index) const
+    {
+        return bitField.IsSet(index + 1);
+    }
+
+    /**
+     * @brief Removes the element at the specified position from the array. If the provided index
+     * is larger than the array's size, an exception will be thrown.
+     * @param index the index of the element we want to remove.
+     */
+    inline void Remove(const size_t& index)
+    {
+        bitField.UnsetBit(index + 1);
+        count--;
+    }
+
+    /**
+     * @brief Clears the MSArray.
+     */
+    inline void Clear()
+    {
+        bitField.Clear();
+        count = 0;
+    }
+
+    /**
+     * Creates a base managed iterator for the array. This iterator will ignore elements which
+     * have not been previously assigned. This method is slower than the FIterator but ensures
+     * no garbage data is accessed
+     * @return a MIter to iterate over the array
+     */
+    inline Utils::MIter<MSArray<T, S>, T> Iterator()
+    {
+        return {this};
+    }
+
+    /**
+     * Creates a fast iterator type. The fast iterator iterates over every single element in an
+     * array, regardless of whether they have been assigned to. This method is faster than the
+     * default iterator but is less safe
+     * @return an Iter instance to iterate over the array
+     */
+    Utils::Iter<MSArray<T, S>, T> FIterator()
```

Comment:
Prefix with "Create" I reckon. It's not clear that it will create a new object otherwise.

Reply:
I'm not the biggest fan of it because it makes any iteration feel really bulky. Regardless I've changed it back.